.github/workflows/ci-test.yml

```diff
@@ -1,20 +1,42 @@
 name: CI Test
 
-on: [push, pull_request]
+on: [pull_request]
 
 jobs:
   test:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [12.x]
     steps:
-    - uses: actions/checkout@v1
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - name: Install dependencies
-      run: npm ci
-    - name: Run tests
-      run: npm test
+      # setup
+      - uses: actions/checkout@v2
+      - name: Read .nvmrc
+        run: echo "##[set-output name=NVMRC;]$(cat .nvmrc)"
+        id: nvm
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.nvm.outputs.NVMRC }}"
+      - name: Install dependencies
+        run: npm ci
+
+      # test code
+      - name: Run tests
+        run: npm test
```

Comment:
I'm cool with this not running on `push` now, but thoughts on moving this to a separate, parallel job within this same workflow? I think there is still value in differentiating between unit tests and generated type linting. Thoughts?

Reply:
Why I removed `push`? Because it basically does the same thing as `pull_request`. So for every PR, it was running two time. Maybe I'm missing something here, I remember we did some weird thing because it's a fork..

Regarding tests, we don't actually have many unit tests right now, so our main test is linting generated types :) If we want to differentiate - we always can look into workflow log and see which step failed. Probably when we'll add more unit tests - we can split it into two jobs/workflows.

Reply:
Alright, yeah I mean any reason we wouldn't want to do that now? The jobs would live in this same workflow file, so I think we would still achieve your same goal of consolidation.

Happy to make the change if you're not opposed.

Reply:
Well, I don't like that we have to copy-paste the setup steps for each job, it's kinda odd and I couldn't find a way to reuse steps between jobs.
Having them in the same file will be better than having two different workflows as it was before.
Also, we probably don't want to lint types if tests failed.
Making it run in parallel would not benefit us because tests run in seconds and linting runs in minutes.
Probably I just don't see why separating tests from linting in two different jobs is good in our case... Could you explain to me why you want to do it?

Reply:
It's really just to explicitly differentiate between unit tests and generated type linting. Let's just leave it for now.